docs/guides/features/ipsec-guide.md

```diff
@@ -0,0 +1,135 @@
+# IPsec Offload
+
+The Inline IPsec offload feature uses the
+[IPDK Infrastructure Application Interface](https://ipdk.io/documentation/Interfaces/InfraApp/)
+to enable cryptographically-secure data traffic. The IPsec control plane
+(IKE protocol) is offloaded, thus avoiding specialized drivers or control
+planes burdening compute instances. Further details are available on the
+[IPDK Inline Acceleration - IPsec](https://ipdk.io/documentation/Recipes/InlineIPsec/)
+page.
+
+This document is intended to be used in conjunction with the
+[IPsec Recipe](https://github.com/ipdk-io/ipsec-recipe), which provides
+strongSwan plugin code to configure and program control messages into the
+target.
+
+The IPsec offload feature is only supported on the Intel&reg; IPU E2100 target.
+
+## Feature overview
+
+The `infrap4d` process provides the gRPC server-side support for P4RT and
+gNMI messages. strongSwan acts as the gRPC client in this context.
+Alternatively, you can use other clients that implement the IKE stack. The
+client communicates using P4RT to program Security Policy Database (SPD) and
+uses gNMI to configure Security Association Database (SAD) entries, which
+includes encryption keys, re-keying etc.
+
+The [openconfig-ipsec-offload](https://github.com/ipdk-io/openconfig-public/blob/master/release/models/ipsec/openconfig-ipsec-offload.yang)
+YANG model is used to configure IPsec offload.
+
```

Comment:
The switch from header level two to header level three doesn't make sense. The following paragraphs aren't part of the Feature Overview, they represent a new topic, which is a How-To.

My suggestion would be to insert a level two heading at this point.

## Enabling IPsec

Follow the sequence of steps . . .